pkg/server/api.go

```diff
@@ -46,6 +47,9 @@ func (a *APIServer) Serve() {
 	mcs := &http.Server{
 		Addr:    fmt.Sprintf(":%v", a.port),
 		Handler: a.handler,
+		TLSConfig: &tls.Config{
+			MinVersion: tls.VersionTLS12,
```

Comment:
So 4 years from now, when the world has moved past TLS 1.2 and  Golang 1.X bumps the minimum to TLS 1.3...we'll start *downgrading* things here.  How do we avoid that?

Maybe something like having a `ClusterTlsMinVersion` constant in openshift/api?

Reply:
That will impact openshift as a whole I guess but it sounds like future proof enough to me - there are other components running a server (right?)

Reply:
I'll create an issue for this so we don't forget